src/providers/seventv/SeventvEmotes.cpp

```diff
@@ -24,12 +24,20 @@ namespace {
     // maximum pageSize that 7tv's API accepts
     constexpr int maxPageSize = 150;
 
-    Url getEmoteLink(const EmoteId &id, const QString &emoteScale)
+    Url getEmoteLink(const QJsonArray &jsonUrls, const QString &emoteScale)
 
     {
-        const QString urlTemplate("https://cdn.7tv.app/emote/%1/%2");
-
-        return {urlTemplate.arg(id.string, emoteScale)};
+        for (const auto &urlPair : jsonUrls)
+        {
+            // 7tv has its emotes in an array for some reason
+            // The first entry in the array is the scale, second is the url
+            auto urlArray = urlPair.toArray();
+            if (urlArray.size() == 2 && urlArray.at(0).toString() == emoteScale)
+            {
+                return {urlArray.at(1).toString()};
+            }
+        }
+        return {""};
```

Comment:
Scheme for emote URLs hasn't changed, why do we change this function to a very confusing syntax instead of just using emote IDs? Just assuming array entries based on their indexes and then comparing with hardcoded value passed through `emoteScale` seems like a clumsy solution. I'd honestly much rather prefer old solution or alternatively, get rid of this function and make an `ImageSet` consisting of `Image`s from a for loop or something. Maybe I'm too picky, no idea.

Reply:
This was requested here https://github.com/Chatterino/chatterino2/pull/3071#issuecomment-1008346507 by @AnatoleAM to future proof it to different url structures. This loop is required since the actual API object isn't a map (it really should be).

Reply:
It has changed to a different structure in API V3 for the record.

Reply:
Fair enough, sorry for my excessive nitpickiness.